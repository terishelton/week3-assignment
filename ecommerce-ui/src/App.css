.App { text-align: center; }

.NotAirBnB { 
    max-width: 80%;
    margin: 0 auto;
}

ul { list-style-type: none; }

#rentals, .CartRental {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-column-gap: 10px;
}

.CartRental { 
    max-width: 70%;
    margin: 0 auto 10px;
    grid-template-columns: .5fr .2fr .2fr; 
    text-align: left;
    align-items: center;
}

.CartTitle { font-weight: bold; }

.rental {
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 10px;
}

.rentalImage { 
    width: 100%; 
    height: auto; 
}

.houseType { text-transform: uppercase; }

.CartRental span {
    margin-right: 20px;
}

button {
    background-color: purple;
    border: 1px solid purple;
    color: white;
    padding: 5px 10px;
    border-radius: 5px;
    font-weight: bold;
    font-size: 1em;
}

button:hover, button:active, button:focus {
    background-color: darkcyan;
    border: 1px solid darkcyan;
    cursor: pointer;
}

hr { margin: 50px 0;}