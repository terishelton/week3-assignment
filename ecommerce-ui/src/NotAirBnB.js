import React, { Component } from 'react';
import Rentals from './Rentals';
import Cart from './Cart';
import airbnbs from './airbnbs.json';

class NotAirBnB extends Component {
    state = {
        rentals: airbnbs,
        selectedRentals: [],
        cartTotal: 0
    }
    
    addRental = (index) => {
        const selectedRental = this.state.rentals[index];
        const currentTotal = this.state.cartTotal;
        const rentalCost = selectedRental.payment.cost;
        const newTotal = currentTotal + rentalCost;

        if (this.state.selectedRentals.includes(selectedRental)) {
            alert('You have already selected this rental.');
            return;
        }

        this.setState(prevState => {
            return {
              selectedRentals: [...prevState.selectedRentals, selectedRental],
              cartTotal: newTotal
            };
        });
    }

    
    deleteRental = (index) => {
        const selectedRental = this.state.rentals[index];
        const currentTotal = this.state.cartTotal;
        const rentalCost = selectedRental.payment.cost;
        const newTotal = currentTotal - rentalCost;

        this.setState(prevState => {
            const selectedRentals = [...prevState.selectedRentals];
            selectedRentals.splice(index, 1);
      
            return {
              selectedRentals,
              cartTotal: newTotal
            };
      });
    }

    render() {
        return(
            <div className="NotAirBnB">
                <h1>Not AirBnB</h1>
                <Cart 
                    rentals={this.state.selectedRentals}
                    cartTotal={this.state.cartTotal}
                    onDeleteRental={this.deleteRental}
                />
                <hr />
                <Rentals
                    rentals={this.state.rentals}
                    onAddRental={this.addRental}
                />
            </div>
        );
    }
}

export default NotAirBnB;