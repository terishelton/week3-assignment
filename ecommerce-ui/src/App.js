import React from 'react';
import './App.css';
import NotAirBnB from './NotAirBnB';

function App() {
  return (
    <div className="App">
      <NotAirBnB />
    </div>
  );
}

export default App;
