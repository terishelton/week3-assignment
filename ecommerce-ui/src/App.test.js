import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, render, mount } from 'enzyme';
import App from './App';
import NotAirBnB from './NotAirBnB';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow renders', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.render()).toMatchSnapshot();
});
