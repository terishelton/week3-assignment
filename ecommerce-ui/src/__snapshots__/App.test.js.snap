// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<div
  class="App"
>
  <div
    class="NotAirBnB"
  >
    <h1>
      Not AirBnB
    </h1>
    <div
      id="Cart"
    >
      <h2>
        Your Cart
      </h2>
      <ul />
      <h3>
        Your Total
      </h3>
      <p>
        $0
      </p>
    </div>
    <hr />
    <div
      class="allRentals"
    >
      <h2>
        Available Rentals
      </h2>
      <ul
        id="rentals"
      >
        <li
          class="rental"
        >
          <img
            alt="Centrally-located Manhattan studio"
            class="rentalImage"
            src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
          />
          <p
            class="houseType"
          >
            Apartment
          </p>
          <h3>
            Centrally-located Manhattan studio
          </h3>
          <p
            class="price"
          >
            $105 per person
          </p>
          <p
            class="rating"
          >
            <span>
              3
            </span>
             
            <img
              alt="star"
              src="star.svg"
              width="12"
            />
             
            <span>
              (133)
            </span>
          </p>
          <button>
            Add to Cart
          </button>
        </li>
        <li
          class="rental"
        >
          <img
            alt="Denver apartment outside Union Station"
            class="rentalImage"
            src="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
          />
          <p
            class="houseType"
          >
            Apartment
          </p>
          <h3>
            Denver apartment outside Union Station
          </h3>
          <p
            class="price"
          >
            $130 per person
          </p>
          <p
            class="rating"
          >
            <span>
              4.5
            </span>
             
            <img
              alt="star"
              src="star.svg"
              width="12"
            />
             
            <span>
              (381)
            </span>
          </p>
          <button>
            Add to Cart
          </button>
        </li>
        <li
          class="rental"
        >
          <img
            alt="Quaint 3-bedroom home in Gamla stan"
            class="rentalImage"
            src="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
          />
          <p
            class="houseType"
          >
            Entire House
          </p>
          <h3>
            Quaint 3-bedroom home in Gamla stan
          </h3>
          <p
            class="price"
          >
            $230 per person
          </p>
          <p
            class="rating"
          >
            <span>
              5
            </span>
             
            <img
              alt="star"
              src="star.svg"
              width="12"
            />
             
            <span>
              (83)
            </span>
          </p>
          <button>
            Add to Cart
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
